### PR TITLE
crimson/os/seastore/omap_manager: bug fix for https://tracker.ceph.com/issues/56399

### PR DESCRIPTION
**Pull request by xxhdx1985126**
Fixes: https://tracker.ceph.com/issues/56399
Signed-off-by: Xuehan Xu <xxhdx1985126@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/crimson/os/seastore/omap_manager/btree/btree_omap_manager.cc [hunk 1]
```diff
@@ -24,8 +20,8 @@ BtreeOMapManager::BtreeOMapManager(
 BtreeOMapManager::initialize_omap_ret
 BtreeOMapManager::initialize_omap(Transaction &t, laddr_t hint)
 {
-
-  logger().debug("{}", __func__);
+  LOG_PREFIX(BtreeOMapManager::initialize_omap);
+  DEBUGT("{} hint: {}", t, __func__, hint);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/omap_manager/btree/btree_omap_manager.cc:24]: `__func__` is not necessary.

File: src/crimson/os/seastore/omap_manager/btree/btree_omap_manager.cc [hunk 2]
```diff
@@ -101,7 +101,8 @@ BtreeOMapManager::omap_get_value(
   Transaction &t,
   const std::string &key)
 {
-  logger().debug("{}: {}", __func__, key);
+  LOG_PREFIX(BtreeOMapManager::omap_get_value);
+  DEBUGT("{}: {}", t, __func__, key);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/omap_manager/btree/btree_omap_manager.cc:105]: `__func__` is not necessary.

File: src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc [hunk 3]
```diff
@@ -268,12 +276,15 @@ OMapInnerNode::clear(omap_context_t oc)
 OMapInnerNode::split_children_ret
 OMapInnerNode:: make_split_children(omap_context_t oc)
 {
-  logger().debug("OMapInnerNode: {}", __func__);
+  LOG_PREFIX(OMapInnerNode:: make_split_children);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc:288]: extra space between OMapInnerNode:: and make_split_children

File: src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc [hunk 4]
```diff
@@ -268,12 +276,15 @@ OMapInnerNode::clear(omap_context_t oc)
...
+  DEBUGT("this: {}", oc.t, *this);
   return oc.tm.alloc_extents<OMapInnerNode>(oc.t, oc.hint,
     OMAP_INNER_BLOCK_SIZE, 2)
-    .si_then([this] (auto &&ext_pair) {
+    .si_then([this, oc] (auto &&ext_pair) {
+      LOG_PREFIX(OMapInnerNode:: make_split_children);
```
Comment (cyx1231st) [inline src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc:293]: extra space between OMapInnerNode:: and make_split_children

**Review by cyx1231st — APPROVED** [review]

**Review by liu-chunmei — APPROVED** [review]
LGTM, except for the small issues pointed out by yingxin.

**Review by athanatos — APPROVED** [review]

**Comment by cyx1231st** [comment]
jenkins test make check

**Pull request merged**